### PR TITLE
AY profile generated at the end of installation contains valid hostname

### PR DESCRIPTION
**Pull request by mchf**
Forgot to base previous PR on maintenance branch so this is a cherry pick of https://github.com/yast/yast-yast2/pull/434

### CODE REVIEW

File: library/types/src/modules/Hostname.rb [hunk 1]
```diff
@@ -155,30 +155,20 @@ def MergeFQ(hostname, domain)
...
-                     else
-                       ""
-                     end
-
-        if fqhostname == "" || fqhostname.nil?
-          # last resort (#429792)
-          fqhostname = "linux.#{@DefaultDomain}"
-        end
-        Builtins.y2warning("Using fallback hostname %1", fqhostname)
+        Builtins.y2warning("Using fallback hostname")
```
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: I think at least you should print what return previous command, so it is logged

File: library/types/src/modules/Hostname.rb [hunk 2]
```diff
@@ -155,30 +155,20 @@ def MergeFQ(hostname, domain)
...
+
+        fqhostname = SCR.Read(path(".target.string"), "/etc/hostname") || ""
+        fqhostname = "linux.#{@DefaultDomain}" if fqhostname.empty?
       else
-        fqhostname = Ops.get_string(hostname_data, "stdout", "")
+        fqhostname = hostname_data["stdout"] || ""
```
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: why `""` is needed? I think above you checked that it is valid hostname...also yo remove to_s and strip, so above you check if stripped and converted version is correct and here you do nto use it.

File: library/types/src/modules/Hostname.rb [hunk 3]
```diff
@@ -155,30 +155,20 @@ def MergeFQ(hostname, domain)
...
-                     end
-
-        if fqhostname == "" || fqhostname.nil?
-          # last resort (#429792)
-          fqhostname = "linux.#{@DefaultDomain}"
-        end
-        Builtins.y2warning("Using fallback hostname %1", fqhostname)
+        Builtins.y2warning("Using fallback hostname")
+
+        fqhostname = SCR.Read(path(".target.string"), "/etc/hostname") || ""
```
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: what if file contain some new line? I think at least strip here would be helpful. so maybe something like `SCR.Read(path(".target.string"), "/etc/hostname").to_s.strip`

File: library/types/src/modules/Hostname.rb [hunk 4]
```diff
@@ -155,30 +155,20 @@ def MergeFQ(hostname, domain)
...
+        fqhostname = "linux.#{@DefaultDomain}" if fqhostname.empty?
       else
-        fqhostname = Ops.get_string(hostname_data, "stdout", "")
+        fqhostname = hostname_data["stdout"] || ""
       end
 
       fqhostname = String.FirstChunk(fqhostname, "\n")
```
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: ah and here it do new line cleaning. Interesting.

File: library/types/src/modules/Hostname.rb [hunk 5]
```diff
@@ -247,7 +237,7 @@ def CurrentDomain
     #
     # Basicaly, linuxrc sometimes resolves IP when querying "hostname --fqdn"
     def invalid_hostname?(hostname)
-      Mode.installation && IP.Check(hostname)
+      Stage.initial && IP.Check(hostname)
```
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: so only invalid hostname is when it contain IP? Do not look much robust to me.

File: library/types/src/modules/Hostname.rb (hunk 4 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: yes. Some Ycp relics stayed ... I've focused on workflow (simplifying ```if ... if ... else``` constructs

File: library/types/src/modules/Hostname.rb (hunk 2 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: it doesn't hurt me ... so I put it there automatically.

File: library/types/src/modules/Hostname.rb (hunk 1 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: this is tracing note ... domain is printed bellow.

Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: domain but not output of this command which can be useful.

File: library/types/src/modules/Hostname.rb (hunk 2 above)
Comment (jreidinger) [inline library/types/src/modules/Hostname.rb]: well, it confuse reader as it looks like you expected nil there. So use it only when needed :) Do not confuse network reader more then needed.

Comment (mchf) [inline library/types/src/modules/Hostname.rb]: Well I think that there are much more things which can confuse reader of network code ;-) But ok I'll drop it.

File: library/types/src/modules/Hostname.rb (hunk 5 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: this method is a workaround / hotfix until fate#900647 is somehow handled.

File: library/types/src/modules/Hostname.rb (hunk 1 above)
Comment (mchf) [inline library/types/src/modules/Hostname.rb]: well I usually agree / do so. But in this case the output can be easily extracted from other output and I don't want to fill the log more than needed.

**Comment by jreidinger** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for review.

**Pull request merged**